packages/components/addon/components/hds/button/index.ts

```diff
@@ -13,7 +13,29 @@ export const SIZES = ['small', 'medium', 'large'];
 export const COLORS = ['primary', 'secondary', 'tertiary', 'critical'];
 export const ICONPOSITIONS = ['leading', 'trailing'];
 
-export default class HdsButtonIndexComponent extends Component {
+export interface HdsButtonSignature {
+  Args: {
+    size?: 'small' | 'medium' | 'large';
+    color?: 'primary' | 'secondary' | 'tertiary' | 'critical';
```

Comment:
> Also, in my previous design system, that's how we would have declared the enums:

Defining type for arguments outside the interface definition has the only benefit that these (e.g. `HdsButtonColor `) can potentially be re-used in other components. However, I ended up extracting them outside the interface [while defining union types from the arrays of strings](https://github.com/hashicorp/design-system/pull/1580/commits/fb4c38de9a8041924f57b8c344ba04f8eb51c7dc) – to avoid having to maintain two lists, as you suggested.